utils/utils.go

```diff
@@ -104,3 +107,176 @@ func HandlePanic() {
 	}
 }
+
+// It checks if the cmd is related to docker or not, it also returns if its a docker compose file
+func IsDockerRelatedCmd(cmd string) (bool, string) {
+	// Check for Docker command patterns
+	dockerCommandPatterns := []string{
+		"docker-compose ",
+		"sudo docker-compose ",
+		"docker compose ",
+		"sudo docker compose ",
+		"docker ",
+		"sudo docker ",
+	}
+
+	for _, pattern := range dockerCommandPatterns {
+		if strings.HasPrefix(strings.ToLower(cmd), pattern) {
+			if strings.Contains(pattern, "compose") {
+				return true, "docker-compose"
+			}
+			return true, "docker"
+		}
+	}
+
+	// Check for Docker Compose file extension
+	dockerComposeFileExtensions := []string{".yaml", ".yml"}
+	for _, extension := range dockerComposeFileExtensions {
+		if strings.HasSuffix(strings.ToLower(cmd), extension) {
+			return true, "docker-compose"
+		}
+	}
+
+	return false, ""
+}
+
+type RecordFlags struct {
+	Path             string
+	Command          string
+	ContainerName    string
+	Proxyport        uint32
+	NetworkName      string
+	Delay            uint64
+	BuildDelay       time.Duration
+	PassThroughPorts []uint
+	ConfigPath       string
+	EnableTele       bool
+}
+
+type TestFlags struct {
+	Path               string
+	Proxyport          uint32
+	Command            string
+	Testsets           []string
+	ContainerName      string
+	NetworkName        string
+	Delay              uint64
+	BuildDelay         time.Duration
+	ApiTimeout         uint64
+	PassThroughPorts   []uint
+	ConfigPath         string
+	MongoPassword      string
+	CoverageReportPath string
+	EnableTele         bool
+	WithCoverage       bool
+}
+
+func getAlias(keployAlias *string, logger *zap.Logger) {
+	// Get the name of the operating system.
+	osName := runtime.GOOS
+	if osName == "Windows" {
+		logger.Error("Windows is not supported. Use WSL2 instead.")
+		return
+	}
+	if osName == "darwin" {
+		//Get the current docker context.
+		cmd := exec.Command("docker", "context", "ls", "--format", "{{.Name}}")
+		out, err := cmd.Output()
+		if err != nil {
+			logger.Error("Failed to get the current docker context", zap.Error(err))
+			return
+		}
+		dockerContext := strings.Split(strings.TrimSpace(string(out)), "\n")[0]
+		dockerContext = strings.Split(dockerContext, "\n")[0]
+		if dockerContext == "colima" {
+			logger.Info("Starting keploy in docker with colima context, as that is the current context.")
+		}else{
+			logger.Info("Starting keploy in docker with default context, as that is the current context.")
+		}
+		*keployAlias = "sudo docker run --pull always --name keploy-v2 -e BINARY_TO_DOCKER=true -p 16789:16789 --privileged --pid=host -it -v " + os.Getenv("PWD") + ":/files -v /sys/fs/cgroup:/sys/fs/cgroup -v debugfs:/sys/kernel/debug:rw -v /sys/fs/bpf:/sys/fs/bpf -v /var/run/docker.sock:/var/run/docker.sock -v " + os.Getenv("HOME") + "/.keploy-config:/root/.keploy-config -v " + os.Getenv("HOME") + "/.keploy:/root/.keploy --rm ghcr.io/keploy/keploy "
```

Comment:
In the case of default docker context, the alias is correct but for colima, the alias is the same as for Linux.

Reply:
done